File: src/main/java/net/jgp/books/spark/ch13/lab999_functions/AddMonthsApp.java
package net.jgp.books.spark.ch13.lab999_functions;

import static org.apache.spark.sql.functions.add_months;
import static org.apache.spark.sql.functions.col;

import org.apache.spark.sql.Dataset;
import org.apache.spark.sql.Row;
import org.apache.spark.sql.SparkSession;

/**
 * add_months function.
 * 
 * @author jgp
 */
public class AddMonthsApp {

  public static void main(String[] args) {
    AddMonthsApp app = new AddMonthsApp();
    app.start();
  }

  /**
   * The processing code.
   */
  private void start() {
    SparkSession spark = SparkSession.builder()
        .appName("abs function")
        .master("local[*]")
        .getOrCreate();

    Dataset<Row> df = spark.read().format("csv")
        .option("header", true)
        .option("imferSchema", true)
        .load("data/functions/dates.csv");

    df = df
        .withColumn("add_months+2", add_months(col("date_time"), 2))
        .withColumn("add_months+val",
            add_months(col("date_time"), col("val")));

    df.show(5, false);
    df.printSchema();
  }
}

File: src/main/python/lab999_functions/addMonthsApp.py
"""
 add_months function.

 @author rambabu.posa
"""
from pyspark.sql import SparkSession
from pyspark.sql import (functions as F)
import os

current_dir = os.path.dirname(__file__)
relative_path = "../../../../data/functions/dates.csv"
absolute_file_path = os.path.join(current_dir, relative_path)

# Creates a session on a local master
spark = SparkSession.builder.appName("add_months function") \
    .master("local[*]").getOrCreate()

df = spark.read.format("csv") \
    .option("header", True) \
    .option("imferSchema", True) \
    .load(absolute_file_path)

df = df.withColumn("add_months+2", F.add_months(F.col("date_time"), 2))
# As of now, no support for this operation in PySpark API
# df = df.withColumn("add_months+val", F.add_months(F.col("date_time"), F.col("val")))

df.show(5, False)
df.printSchema()

df.show(5)

spark.stop()